ast-nodes/type-conversion.dart

```diff
@@ -0,0 +1,33 @@
+import '../print-utils.dart';
+import '../symbol-table/scope-element.dart';
+import 'index.dart';
+
+class TypeConversion implements Expression {
+  Expression expression;
+  VarType resultType;
+  bool isConstant;
+  ScopeElement scopeMark;
+
+  TypeConversion(this.expression, this.resultType);
+
+  @override
+  void checkSemantics() {}
+
+  @override
+  void propagateScopeMark(ScopeElement parentMark) {
+    this.scopeMark = parentMark;
+    this.expression.propagateScopeMark(parentMark);
+  }
+
+  @override
+  String toString({int depth = 0, String prefix = ''}) {
+    return (drawDepth(
+            '${prefix}TypeConversion: (${this.resultType.toString()})', depth) +
```

Comment:
I think it's better to have `'${prefix}TypeConversion(${this.resultType.runtimeType})`, because the result of `.toString()` is multiline, it will mess up the output